### ISSUE TITLE
IE 11 not supported

### ISSUE THREAD

**Issue by blackbing** [lovasoa/react-contenteditable#43]
related issue: 

- https://github.com/lovasoa/react-contenteditable/issues/19
- https://github.com/facebook/react/issues/7280

**Comment by blackbing** [comment]
send pr for solving this. https://github.com/lovasoa/react-contenteditable/pull/44

**Comment by blackbing** [comment]
@lovasoa I am looking for the IE support solution for this.  

* https://connect.microsoft.com/IE/feedback/details/794285/ie10-11-input-event-does-not-fire-on-div-with-contenteditable-set

Here is a simple idea about solving this problem. 
1. use keyup event instead if IE browser.
2. also handle paste event.

And you mentioned `We can simply shadow the target if it can't be overwritten.` in [here](https://github.com/lovasoa/react-contenteditable/pull/44#issuecomment-262903901). Can you give me some hint about this? I'll make PR ASAP because my project need this support IE.

**Comment by stahlmanDesign** [comment]
I don't know if this helps but on some React input components I use `onInput` instead of `onChange` so that IE11 behaves like all other browsers. Apparently onChange will work as expected in IE when Fiber a.k.a React v16.0 comes out.

**Comment by slorber** [comment]
@stahlmanDesign can you provide an example implementation of your solution please?

**Comment by stahlmanDesign** [comment]
Replace `onChange` with `onInput`
http://codepen.io/stahlmandesign/pen/JWpNKJ

**Comment by slorber** [comment]
@stahlmanDesign I don't understand, the project is about a contentEditable div, not a regular input.
Also onInput is already what is used: https://github.com/lovasoa/react-contenteditable/blob/master/src/react-contenteditable.js#L17

**Comment by guillaumepotier** [comment]
Hi there,

What is the status of this one, is this still an open issue for IE11?

Thanks

**Comment by slorber** [comment]
@guillaumepotier this is supposed to be resolved in React 15.6: 

https://facebook.github.io/react/blog/2017/06/13/react-v15.6.0.html
https://github.com/facebook/react/pull/8575

**Comment by guillaumepotier** [comment]
How, that's a good news! I'll test it then when I'll update it. Thanks for your help

**Comment by christianharke** [comment]
@slorber I just updated to react 15.6.1, but unfortunately, the onChange prop function still doesn't get called in IE11. I hope I'm just doing something wrong...

**Comment by slorber** [comment]
@christianharke I'm not maintaining this project nor I have upgraded to 15.6.1, if you think it does not work try to reproduce with raw react div and if it does not work fill a bug in react repo

**Comment by notatestuser** [comment]
@blackbing @guillaumepotier @lovasoa We are using react v15.6.1 and still had problems with this component in IE11.

I've successfully modified this component to use a MutationObserver with working IE11 support: https://github.com/e1-bsd/omni-common-ui/pull/35/commits/a6031f2db6320e6b85dec60cc1adfb1791ffae56

**Comment by MartinDawson** [comment]
This is broken still in IE11 for React 16.

**Comment by MartinDawson** [comment]
Seems to work fine now.

**Comment by guillaumepotier** [comment]
I confirm this is still broken on IE11 with React and React-dom 1.16.3

Tested @notatestuser change and it works. Could we consider integrating it?

**Comment by mccambridge** [comment]
Is this repo still supported? Not working on IE 11 for me.

**Comment by lovasoa** [comment]
Yes, this repo is still maintained. I'd prefer not to merge @notatestuser PR because it breaks the current API. If onInput is not fired, then react should fix it.  The issue was closed in react. If it is still present, then they should fix it. Please go complain there :) https://github.com/facebook/react/issues/7280

**Comment by slgauravsharma** [comment]
@lovasoa **IE11 Support** if we add onKeyUp props. so can i sent pull request for this fix ?

**Comment by mdshohelrana** [comment]
Hi there,

What is the status of this one, is this still an open issue for IE11?

Thanks
